python/dgl/graphbolt/feature_store.py

```diff
@@ -26,6 +26,25 @@ def read(self, ids: torch.Tensor = None):
         """
         raise NotImplementedError
 
+    def read_async(self, ids: torch.Tensor):
+        """Read the feature by index asynchronously.
+        Parameters
+        ----------
+        ids : torch.Tensor
+            The index of the feature. Only the specified indices of the
+            feature are read.
+        Returns
+        -------
+        A future containing torch.Tensor. It can be accessed by calling
+        `.wait()`. on the returned object.
+            The future containing the read feature.
+        """
+        raise NotImplementedError
+
+    def read_async_num_stages(self, ids_device: torch.device):
+        """The number of stages of the read_async operation."""
+        raise NotImplementedError
+
```

Comment:
What do you think about this @frozenbugs, this will be the first PR when I split it.